app/services/metrics/review_turnaround_per_project_processor.rb

```diff
@@ -0,0 +1,120 @@
+  #
+  #   up to
+  #
+  #   - the instant when a PullRequestReviewEvent for that same PullRequestReview
+  #     is 'submitted', 'edited' or 'dismissed'.
+  #     (https://developer.github.com/v3/activity/events/types/#pullrequestreviewevent)
+  #
+  # This processor generates a review_turnaround metric for:
+  #     - each project
+  #     in
+  #     - a time interval (like 'daily', 'weekly', 'all_times')
+  #
+  # For example, given two projects A and B and a daily interval the
+  # ReviewTurnaroundPerProjectProcessor would generate the following metrics:
+  #
+  #     - project A, 2020-01-01 => 100
+  #     - project B, 2020-01-01 => 20
+  #     - project A, 2020-01-02 => 130
+  #     - project B, 2020-01-02 => 120
+  #
+  # For the same project and a monthly interval generates the following metrics
+  #
+  #     - project A, 2020-01-01 => 320
+  #     - project B, 2020-01-01 => 140
+  #     - project A, 2020-02-00 => 150
+  #     - project B, 2020-02-00 => 110
+  #
+  # Therefore to generate the metrics from a collection of Github events the
+  # ReviewTurnaroundPerProjectProcessor requires the following parameters:
+  #
+  #     - an interval of time (ej. 1 month)
+  #     - the initial point in time for the interval (ej. 2020-01-01)
+  #     - the collection of events reveived in that interval
+  class ReviewTurnaroundPerProjectProcessor < BaseMetricProcessor
+    private
+
+    ##
+    # review_turnaround_per_project: { project_id: review_turnaround_value }
+    #   Keeps the values of the review_turnaround for each project.
+    #   Allows to calculate the metrics for all the projects in a single pass.
+    #
+    # pull_request_reviewed: { pull_request_id: true|nil }
+    #   A PullRequest can have many Review events. The
+    #   review_turnaround only considers the first Review. This flag is set
+    #   to true if a previous Review event was the first Review to ignore all
+    #   the Review events after that one.
+    def create_accumulators
+      {
+        review_turnaround_per_project: Hash.new { |hash, key| hash[key] = [] },
+        pull_request_reviewed: {}
+      }
+    end
+
+    def process_event(event:, accumulators:)
+      pull_request_id = event.data['pull_request']['id']
+
+      review_turnaround_value = review_turnaround_as_seconds(event: event)
+      accumulators[:review_turnaround_per_project][event.project.name] << review_turnaround_value
+      accumulators[:pull_request_reviewed][pull_request_id] = :reviewed
+    end
+
+    ##
+    # Updates the metrics for all the project in the given review_turnaround_per_project
+    def update_metrics(time_interval:, accumulators:)
+      accumulators[:review_turnaround_per_project].each_pair do |project_id, values|
+        average_value = values.sum / values.size
+
+        update_metric(
+          entity_key: project_id,
+          metric_key: 'review_turnaround',
+          value: average_value,
```

Comment:
It'd be interesting to have the avg, min, max, median. Have you already discussed about this? otherwise we can start the conversation as something to do in the future